### ISSUE TITLE
Feature request: more specific events for feature changes

### ISSUE THREAD

**Issue by ffxsam** [openlayers/openlayers#8650]
Right now I'm using the `change` event for features, to detect when the user modifies a polygon on the map. Unfortunately, `change` is a catch-all for any changes to features, including style changes, ID changes, etc. It would be nice to have some way to tell what type of modification triggered the change event.

My hacky workaround is to use this function to make feature changes without triggering my `change` event handler:

```js
function silentChange(feature, func) {
  feature.removeEventListener('change', this.debouncedUpdateFeature);
  func();
  feature.addEventListener('change', this.debouncedUpdateFeature);
}
```

**Comment by fredj** [comment]
Instead of listening to the feature, can't you listen to the geometry instead? Something like:
```js
feature.getGeometry().addEventListener('change', this.onGeometryChange);
```

**Comment by ffxsam** [comment]
For some reason, I didn't even think of doing that. I guess I missed that line in the API docs:

>To get notified of changes to the geometry, register a listener for the generic change event on your geometry instance.

It would be nice to have sections in the API reference for each class, e.g. "Methods" (which already exists), "Events", etc.

**Comment by fredj** [comment]
FYI, the api documentation is currently broken and will be fixed soon, see  #8449
